primitives/runtime/src/traits.rs

```diff
@@ -1217,6 +1217,8 @@ macro_rules! impl_opaque_keys {
 			)*
 		}
 	) => {
+		#[cfg(feature = "std")]
+		use $crate::serde as __opaque_keys_serde_import__;
```

Comment:
the polkadot build complains: the name `__opaque_keys_serde_import__` is defined multiple times

But if we abuse from paste featureto concatenate string in doc, we can do:

```diff
diff --git a/primitives/runtime/src/traits.rs b/primitives/runtime/src/traits.rs
index f95a7aa200..bdd5efde59 100644
--- a/primitives/runtime/src/traits.rs
+++ b/primitives/runtime/src/traits.rs
@@ -1216,9 +1216,36 @@ macro_rules! impl_opaque_keys {
                                pub $field:ident: $type:ty,
                        )*
                }
+       ) => {
+               $crate::paste::paste!(
+                       $crate::impl_opaque_keys!(@with_serde_crate_literal
+                               $( #[ $attr ] )*
+                               pub struct $name {
+                                       $(
+                                               $( #[ $inner_attr ] )*
+                                               pub $field: $type,
+                                       )*
+                               }
+                               // NOTE: we abuse from `paste` feature which concatenate string in docs to build
+                               // our string
+                               #[doc = "__opaque_keys_serde_import__" $name ""]
+                       );
+               );
+       };
+       (@with_serde_crate_literal
+               $( #[ $attr:meta ] )*
+               pub struct $name:ident {
+                       $(
+                               $( #[ $inner_attr:meta ] )*
+                               pub $field:ident: $type:ty,
+                       )*
+               }
+               #[ doc = $serde_crate_literal:expr ]
        ) => {
                #[cfg(feature = "std")]
-               use $crate::serde as __opaque_keys_serde_import__;
+               $crate::paste::paste!(
+                       use $crate::serde as [< __opaque_keys_serde_import__ $name >];
+               );
                $( #[ $attr ] )*
                #[derive(
                        Default, Clone, PartialEq, Eq,
@@ -1227,7 +1254,7 @@ macro_rules! impl_opaque_keys {
                        $crate::RuntimeDebug,
                )]
                #[cfg_attr(feature = "std", derive($crate::serde::Serialize, $crate::serde::Deserialize))]
-               #[cfg_attr(feature = "std", serde(crate = "__opaque_keys_serde_import__"))]
+               #[cfg_attr(feature = "std", serde(crate = $serde_crate_literal))]
                pub struct $name {
                        $(
                                $( #[ $inner_attr ] )*
```

Reply:
Ty ❤️

Reply:
I also thought about something similar 😬